frontend/web/components/modals/CreateSegment.tsx

```diff
@@ -282,16 +276,33 @@ const CreateSegment: FC<CreateSegmentType> = ({
     if (createSuccess && createSegmentData) {
       setSegment(createSegmentData)
       onComplete?.(createSegmentData)
+      setShouldUpdateIdentities(true)
```

Comment:
Adding this logic was necessary due to a data propagation delay. After successfully updating/creating a segment it takes hundreds of milliseconds for `/segments/?identity={ID}` to return identities with the latest segment changes.

Reply:
This is likely due to the time it takes to update the environment document in DynamoDB (which is handled by the task processor). In normal circumstances, this should be less than 1 second, yep.

Reply:
I guess in this case the easy approach is fetching `/segments/?identity={ID}` every time the user tab is rendered, being more intuitive for the user to understand that they can "refresh" the status by switching tabs

Reply:
Yeah, that makes sense.